lib/elixir/lib/exception.ex

```diff
@@ -637,8 +637,16 @@ defmodule UndefinedFunctionError do
   end
 
   def message(%{reason: :"function not exported",  module: module, function: function, arity: arity, exports: exports}) do
-    "function " <> Exception.format_mfa(module, function, arity) <>
-    " is undefined or private" <> did_you_mean(module, function, arity, exports)
+    prefix = "function " <> Exception.format_mfa(module, function, arity) <> " is undefined or private"
+    suffix =
+      if macro_with_same_mfa?(module, function, arity) do
```

Comment:
This is going to end up loading the module now right? When this gets used from Xref that would not turn out well for modules with `:autoload` disabled. I think this condition should maybe be `macro_with_same_mfa?(module, function, arity) and !exports`? Not sure.